server/src/main/java/io/spine/examples/kanban/server/board/BoardAggregate.java

```diff
@@ -46,43 +53,137 @@
+                .build();
+    }
+
+    private ColumnPlaced placeColumn(PlaceColumn c) {
+        ColumnPosition actualPosition =
+                ColumnPosition.newBuilder()
+                              .setIndex(c.getDesiredPosition().getIndex())
+                              .setOfTotal(incrementColumnCount())
+                              .vBuild();
+
         return ColumnPlaced
                 .newBuilder()
-                .setBoard(event.getBoard())
-                .setColumn(event.getColumn())
+                .setBoard(c.getBoard())
+                .setColumn(c.getColumn())
+                .setDesiredPosition(c.getDesiredPosition())
+                .setActualPosition(actualPosition)
                 .vBuild();
     }
 
+    private int incrementColumnCount() {
+        return state().getColumnCount() + 1;
+    }
+
+    private ImmutableList<ColumnMoved> updateTotals() {
+        int currentTotal = state().getColumnCount();
+        int newTotal = incrementColumnCount();
+        ImmutableList.Builder<ColumnMoved> columnsMoved = new ImmutableList.Builder<>();
+
+        for (int i = 1; i <= currentTotal; i++) {
+            columnsMoved.add(updateTotal(i, currentTotal, newTotal));
+        }
+
+        return columnsMoved.build();
+    }
+
+    private ColumnMoved updateTotal(int index, int currentTotal, int newTotal) {
+        ColumnPosition from =
+                ColumnPosition.newBuilder()
+                              .setIndex(index)
+                              .setOfTotal(currentTotal)
+                              .vBuild();
+        ColumnPosition to =
+                ColumnPosition.newBuilder()
+                              .setIndex(index)
+                              .setOfTotal(newTotal)
+                              .vBuild();
+        ColumnId column = state().getColumn(from.zeroBasedIndex());
+
+        return ColumnMoved
+                .newBuilder()
+                .setColumn(column)
+                .setFrom(from)
+                .setTo(to)
+                .vBuild();
+    }
+
+    @Apply
+    private void apply(ColumnPlaced e) {
+        builder().addColumn(e.getActualPosition().zeroBasedIndex(), e.getColumn());
+    }
+
     @Apply
-    private void event(ColumnPlaced e) {
-        builder().setId(e.getBoard())
-                 .addColumn(e.getColumn());
+    private void apply(ColumnMoved e) {
+        builder().removeColumn(e.getFrom().zeroBasedIndex())
+                 .addColumn(e.getTo().zeroBasedIndex(), e.getColumn());
     }
 
     /**
      * Whenever a card created, it is placed to the first column of the board.
      *
-     * @implNote This board knows its columns. So the board listens to the events on new card
-     *           creation, and emits the event with the references to the created card and
-     *           the first column on which the card is to be placed.
+     * @implNote This board knows its columns. So the board listens to the events
```

Comment:
Please split this description into more sentences. It's really difficult to comprehend right now.

Another option is to delete this `@implNote` altogether. It is not helpful right now anyway.